### ISSUE TITLE
hubzilla upgrade --> errors in photo processing

### ISSUE THREAD

**Issue by lazyadmin111** [redmatrix/hubzilla#902]
Hey there,

I updated hubzilla from 2.6.3 to 2.8, and often when I now upload an image it is uploaded and can be accessed and attached to posts as file, but it doesn't show up in photos and can't be embedded.

the debug log also shows this:

> (
>     [0] => 2448
>     [1] => 3264
>     [2] => 2
>     [3] => width="2448" height="3264"
>     [bits] => 8
>     [channels] => 3
>     [mime] => image/jpeg
> )
> 
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:1157:attach_mkdirp: attach_mkdirp: basepath: store/XXXuser1
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:989:attach_mkdir: attach_mkdir: basepath: store/XXXuser1
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:989:attach_mkdir: attach_mkdir: basepath: store/XXXuser1
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:989:attach_mkdir: attach_mkdir: basepath: store/XXXuser1
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:1487:find_folder_hash_by_attach_hash: attach_hash: 8cb70b907103d8798a06555c30127b0038144223097a32c0b9e45ad9cfa38b99
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:1487:find_folder_hash_by_attach_hash: attach_hash: 87dc98c8c8ab7791387812146cba22ee8d49f894bd2d9af00ecbe1474a18b3fb
> 2017-11-03T18:18:21Z:LOG_INFO:9118g8qvkg0feaidjcpnlnn3b6:attach.php:1487:find_folder_hash_by_attach_hash: attach_hash: ffe5d6e3c0295411d1d6c342b0386209f9ca5e0305916bb2ca1a40d34838c740
> 2017-10-30T12:44:39Z:LOG_INFO:cgbrootopac09c1q0ckolmf0m2:photos.php:157:photo_upload: photo_upload: unable to process image

Sometimes however, it works. It seems to be a bit related to which photo is uploaded. Small photos work, but those e.g. more than 1MB in size don#t work. Maybe then there are no thumbnails generated or sth. like that... ?!

Anyone who can tell me how to solve it? -- i can't embed any photos in my post! :-((

**Comment by lazyadmin111** [comment]
I just reverted this commit:
parent bdecb61 commit ea4d1b5101022f928336a20f193f8f73adb4d5a7 @zotlabs zotlabs

in include/photos.php and it seems to work... i will test further now

**Comment by git-marijus** [comment]
You should probably just install imagemagick (not the php extension the programm itself) and point to the `convert` (probably /usr/bin/convert) binary in /admin/site > advanced section (this is new in 2.8).

**Comment by zotlabs** [comment]
It seems there was a bug where the thumbnails might not be generated if it's a large file **and** you don't have the imagick convert installed. I've just fixed this and it should migrate to hubzilla in the next day or so. 

A short term fix is to install imagick and set the convert path in your admin settings (or revert the imagick commit as you've done, which should work for pics from older phones/cameras). We'll try and get the fix pushed at as soon as possible. If your photos came from a newer iphone or android they are simply to large to generate the thumbnails in memory. That's why this code was added.

**Comment by git-marijus** [comment]
I think i see what is going on.
We should probably change this block:
```
	if($args['os_syspath'] && $args['getimagesize']) {
		if($args['getimagesize'][0] > 1600 || $args['getimagesize'][1] > 1600) {
			$imagick_path = get_config('system','imagick_convert_path');
			if($imagick_path && @file_exists($imagick_path)) {
				$tmp_name = $args['os_syspath'] . '-001';
				$newsize = photo_calculate_1600_scale($args['getimagesize']);
				exec($imagick_path . ' ' . $args['os_syspath'] . ' -resize ' . $newsize . '^ ' . $tmp_name);
				$imagedata = @file_get_contents($tmp_name);
				@unlink($tmp_name);
			}
		}
		else {
			$imagedata = @file_get_contents($args['os_syspath']);
		}
		$filename = $args['filename'];
		$filesize = strlen($imagedata);
		// this is going to be deleted if it exists
		$src = '/tmp/deletemenow';
		$type = $args['getimagesize']['mime'];
		$os_storage = 1;
	}
```
to look like that:
```
	if($args['os_syspath'] && $args['getimagesize']) {
		$imagick_path = get_config('system','imagick_convert_path');
		if($imagick_path && ($args['getimagesize'][0] > 1600 || $args['getimagesize'][1] > 1600)) {
			if($imagick_path && @file_exists($imagick_path)) {
				$tmp_name = $args['os_syspath'] . '-001';
				$newsize = photo_calculate_1600_scale($args['getimagesize']);
				exec($imagick_path . ' ' . $args['os_syspath'] . ' -resize ' . $newsize . '^ ' . $tmp_name);
				$imagedata = @file_get_contents($tmp_name);
				@unlink($tmp_name);
			}
		}
		else {
			$imagedata = @file_get_contents($args['os_syspath']);
		}
		$filename = $args['filename'];
		$filesize = strlen($imagedata);
		// this is going to be deleted if it exists
		$src = '/tmp/deletemenow';
		$type = $args['getimagesize']['mime'];
		$os_storage = 1;
	}
```
This way we would still use php-gd on big images if imagemagick is not configured or installed...

**Comment by git-marijus** [comment]
fixed in 6e48c365191e750941b5a9839ba9940242c9f313
